core/src/com/unciv/ui/tilegroups/TileGroupIcons.kt

```diff
@@ -99,13 +100,20 @@ class TileGroupIcons(val tileGroup: TileGroup) {
                 newImage.addActor(holder)
             }
 
+            /*
+            // HOW TO HANDLE WITH NEW ICON OPACITY CONTROL?
             // Instead of fading out the entire unit with its background, we just fade out its central icon,
             // that way it remains much more visible on the map
             if (!unit.isIdle() && unit.civInfo == viewingCiv) {
                 newImage.unitBaseImage.color.a = 0.5f
                 newImage.actionGroup?.color?.a = 0.5f
             }
+             */
+
```

Comment:
Probably the most straightforward approach would be to move the new code above this section then do
`newImage.unitBaseImage.color.a *= 0.5f`
`newImage.actionGroup?.color?.a *= 0.5f`
Users with fully opaque icons would still see the semi-transparent icons once they finish moving a unit, although I don't know if a user with very transparent icons to begin with could notice a difference.